controllers/useraccount/console_usersettings_test.go

```diff
@@ -0,0 +1,285 @@
+package useraccount
+
+import (
+	"context"
+	"fmt"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+func TestDeleteConsoleSettingObjects(t *testing.T) {
+	t.Run("Delete ConfigMap", func(t *testing.T) {
+		t.Run("Configmap found by name and deleted", func(t *testing.T) {
+			ctx := context.Background()
+			cm := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-johnsmith",
+					Namespace: UserSettingNS,
+				},
+			}
+			cl := fake.NewClientBuilder().WithObjects(cm).Build()
+
+			err := deleteResource(ctx, cl, "johnsmith", cm)
+			require.NoError(t, err)
+			// check that the configmap doesn't exist anymore
+			err = cl.Get(ctx, client.ObjectKey{Name: "user-settings-johnsmith", Namespace: UserSettingNS}, &corev1.ConfigMap{})
+			require.True(t, errors.IsNotFound(err))
+		})
+		t.Run("Configmap found by label and deletes successfully", func(t *testing.T) {
+			cm := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-name-no-match",
+					Namespace: UserSettingNS,
+					Labels: map[string]string{
+						ConsoleUserSettingsIdentifier: "true",
+						ConsoleUserSettingsUID:        "johnsmith",
+					},
+				},
+			}
+			cl := fake.NewClientBuilder().WithObjects(cm).Build()
+			err := deleteResource(context.TODO(), cl, "johnsmith", cm)
+			require.NoError(t, err)
+			// check that the configmap doesn't exist anymore
+			err = cl.Get(context.TODO(), client.ObjectKey{Name: "user-settings-name-no-match", Namespace: UserSettingNS}, &corev1.ConfigMap{})
+			require.True(t, errors.IsNotFound(err))
+		})
+		t.Run("multiple configmaps found by label and deletes successfully", func(t *testing.T) {
+			cm1 := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-name-no-match",
+					Namespace: UserSettingNS,
+					Labels: map[string]string{
+						ConsoleUserSettingsIdentifier: "true",
+						ConsoleUserSettingsUID:        "johnsmith",
+					},
+				},
+			}
+			cm2 := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-name-no-match-second",
+					Namespace: UserSettingNS,
+					Labels: map[string]string{
+						ConsoleUserSettingsIdentifier: "true",
+						ConsoleUserSettingsUID:        "johnsmith",
+					},
+				},
+			}
+			cl := fake.NewClientBuilder().WithObjects(cm1, cm2).Build()
```

Comment:
could you please add another object as a noise so we know that the label selection is applied correctly?

Reply:
done in [7b7bd25](https://github.com/codeready-toolchain/member-operator/pull/568/commits/7b7bd259bb5cb92d555a5043585a0d3c06875e8b)